core/shared/src/main/scala/fs2/Stream.scala

```diff
@@ -1722,7 +1729,11 @@ final class Stream[+F[_], +O] private[fs2] (private[fs2] val underlying: Pull[F,
     }.stream
   }
 
-  /** Returns the last element of this stream, if non-empty.
+  /** Returns the last element of this stream, if non-empty and finite.
+    *
+    * Note that this is a fold-right combinator. Pulling from the resulting stream
```

Comment:
Recommend removing "fold-right combinator" as that's not a well known description.